### PR TITLE
implemented voting logic, added optimistic rendering

### PR DESCRIPTION
**Pull request by MrDonev**

### CODE REVIEW

File: src/Components/ReviewCard.jsx [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
+      <div className="ReviewIMG">
+        <img src={reviewObj.review_img_url} alt={reviewObj.title} />
+      </div>
+      <p>Owner: {reviewObj.owner}</p>
+      <p>Category: {reviewObj.category}</p>
+      <p>Created at: {reviewObj.created_at}</p>
+      <p>Title: {reviewObj.title}</p>
+      <p>Comments: {reviewObj.comment_count}</p>
+      <p>Votes {reviewObj.votes}</p>
+      <button
```
Comment (emilyb93) [inline src/Components/ReviewCard.jsx:16]: Refactor this into a \<Link>

File: src/Components/SingleReview.jsx [hunk 2]
```diff
@@ -1,6 +1,18 @@
...
+
+const updateVotes=(value,id,target)=>{
+  value > 0 ? setVotes((votes)=>votes+1) : setVotes((votes)=>votes-1)
+  patchVote(value,id).then(({updatedReview})=>{
+  }).catch((err)=>{
+    value > 0 ? setVotes((votes)=>votes-1) : setVotes((votes)=>votes+1)
+    alert(`There was something wrong with the data! Try again!`)
+  })
+target.disabled=true;
+}
```
Comment (emilyb93) [inline src/Components/SingleReview.jsx:17]: This means you can only click upvote and downvote both once but if you upvoted, and then downvoted, you should be able to vote back up. You either want to give an upvote OR a downvote on something. Try to have some behaviour that enables the upvote button again, if we click the downvote

Comment (emilyb93) [reply src/Components/SingleReview.jsx:17]: Also currently the behaviour of your downvote button, if you have already upvoted, just removes the upvote and doesnt actually down vote

**Review by emilyb93 — CHANGES_REQUESTED** [review]

File: src/Components/ReviewCard.jsx (hunk 1 above)
Comment (MrDonev) [reply src/Components/ReviewCard.jsx:16]: Is there any particular reason to refactor the button to a link? I personally like buttons more :D It was a link in the previous PR but I changed it.

**Pull request merged**